engine/crates/engine-v2/src/request/bind.rs

```diff
@@ -145,39 +173,56 @@ impl<'a> Binder<'a> {
         &self,
         variables: Vec<Positioned<engine_parser::types::VariableDefinition>>,
     ) -> BindResult<Vec<VariableDefinition>> {
+        let mut seen_names = HashSet::new();
         variables
             .into_iter()
             .map(|Positioned { node, .. }| {
-                let name_location = node.name.pos;
                 let name = node.name.node.to_string();
+                let name_location = node.name.pos;
+
+                if seen_names.contains(&name) {
+                    return Err(BindError::DuplicateVariable {
+                        name,
+                        location: name_location,
+                    });
+                }
+                seen_names.insert(name.clone());
```

Comment:
Modifying a mutable variable within a map might be something we want to avoid

Reply:
I could rewrite it as a for, but I didn't see much benefit?